doc/source/serve/autoscaling-guide.md

```diff
@@ -0,0 +1,369 @@
+of requests being processed and waiting in the queue. We recommend you benchmark your application code and set this number based on end to end latency objective.  There is also a maximum limit for the queue that is respected by proxies when assigning requests to replicas, which is defined by `max_concurrent_queries`.
+
+
+:::{note}
+- It is always recommended to load test your workloads. For example, if the use case is latency sensitive, you can lower the `target_num_ongoing_requests_per_replica` number to maintain high performance.
+- Internally, the autoscaler will decide to scale up or down by comparing `target_num_ongoing_requests_per_replica` to the number of `RUNNING` and `PENDING` tasks on each replica.
+- `target_num_ongoing_requests_per_replica` is only a target value used for autoscaling (not a hard limit), the real ongoing requests number can be higher than the config.
+:::
+
+As an example, suppose you have two replicas of a synchronous deployment that has 100ms latency, serving a traffic load of 30 QPS. Then requests will be assigned to replicas faster than the replicas can finish processing them; more and more requests will be queued up at the replica (these are considered "ongoing requests") as time goes on, and so the average number of ongoing requests at each replica will steadily increase. If you set `target_num_ongoing_requests_per_replica = 1`, Serve will detect a higher-than-desired number of ongoing requests per replica, and add more replicas. At 3 replicas, your system would be able to process 30 QPS with 1 ongoing request per replica on average.
+
+For **`target_num_ongoing_requests_per_replica`**, typically we suggest the following.
+* Set `target_num_ongoing_requests_per_replica = 1` if your deployment is synchronous.
+* Set `target_num_ongoing_requests_per_replica > 1` if your deployment is asynchronous.
+
+For **`max_concurrent_queries`**, typically we suggest the following.
+* Set `max_concurrent_queries` to a value between 2 and 5 if the target value is equal to 1.
+* Set `max_concurrent_queries` to ~20 to 50% higher than the target value if the target value is greater than 1.
+
+### Define Upper and Lower Autoscaling Limits
+
+<!-- Now that you understand how Serve autoscales your deployment based on ongoing requests at each replica, you can set the upper and lower autoscaling bounds. Given you have a range you expect traffic to fall in, you can map that to  -->
+
+To use autoscaling, you need to define the minimum and maximum number of resources allowed for your system.
+
+* **`min_replicas`**: This is the minimum number of replicas for the deployment. If you want to ensure your system can deal with a certain level of traffic at all times, you can set `min_replicas` to a positive number. On the other hand, if you anticipate periods of no traffic and want to scale to zero to save costs, you can set `min_replicas = 0`.
```

Comment:
We should mention that the controller and the head node will always be running, even if all the replicas have scaled to zero.